extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/jwt/DefaultJwtDecorator.java

```diff
@@ -46,8 +48,7 @@ public void decorate(JWSHeader.Builder header, JWTClaimsSet.Builder claimsSet) {
                 .issuer(clientId)
                 .subject(clientId)
                 .jwtID(UUID.randomUUID().toString())
-                .notBeforeTime(new Date())
-                .issueTime(new Date())
-                .expirationTime(Date.from(Instant.now().plusSeconds(expiration)));
+                .issueTime(Date.from(clock.instant()))
```

Comment:
nit: consistency - why not use 
`private final Instant createTime = clock.instant();` 
?

Reply:
We want to get the current time every time (used to define token expiration) the method is called, not once when the application starts up